packages/@pollyjs/core/src/test-helpers/jasmine.js

```diff
@@ -0,0 +1,176 @@
+import { afterEach, beforeEach } from './lib';
+
+/**
+ * Flag, showing that Polly is active
+ */
+const IS_POLLY_ACTIVE = Symbol('IS_POLLY_ACTIVE');
```

Comment:
Im not sure what the exact purpose of this symbol is since its always being enabled at the start of the module and disabled at the beginning.

Reply:
This is because of the execution order of the tests. Jasmine (and jest-jasmine2 fork) calls all `describe` and `it` methods before runnings specs, so all `it` calls are using our proxied `it` method if `setupJasmine` was used in the file. This behaviour requires us to somehow tell proxied method should they create new Polly instance or not. 

Consider a test case like this:

```js
describe('describe one', () => {
  setupPolly({/* Defaults */});

  it('test one', () => {
    expect(fetch('/post/1').id).toBe(1);
  });
});

describe('describe two', () => {
  setupPolly({/* Custom options */});

  it('test two', () => {
    expect(fetch('/post/2').statusCode).toBe(404);
  });
});

describe('describe three', () => {
  // Don't want to proxy this one

  it('test three', () => {
    expect(fetch('/post/3').id).toBe(3);
  });
});
```

When jasmine runner executes this file, the order of things will be something like this:

* "describe one" callback is called, proxied methods are not attached yet, they will be attached, before/after hooks are added
* "test one" was called with proxied `it` method 

* "describe two" callback is called, proxied methods are already attached, before/after hooks are added
* "test two" was called with proxied `it` method

* "describe three" callback is called, there is no `setupJasmine`, but Polly is still attached as there are no ways to figure out when to detach it at this point
* "test three" was called with proxied `it` method 

Now the execution of the suites starts.

* "describe one" `beforeAll` hook activates Polly with `IS_POLLY_ACTIVE` flag
* "test one" is executed with Polly
* "describe one" `afterAll` deactivates Polly

* same for "describe two"

* "describe three" doesn't have before/after hooks, so Polly stays inactive and is not attached during the test run

Hope this makes more sense now 😸